tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/interceptor/ftPriorityChange/FaultToleranceInterceptorPriorityChangeAnnotationConfTest.java

```diff
@@ -0,0 +1,104 @@
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+/**
+ * A Client to demonstrate the interceptors ordering behavior of FT and CDI annotations
+ * @author carlosdlr
+ */
+public class FaultToleranceInterceptorPriorityChangeAnnotationConfTest extends Arquillian {
+
+    @Inject
+    private InterceptorComponent testInterceptor;
+
+    @Inject
+    private OrderQueueProducer orderFactory;
+
+    @Deployment
+    public static WebArchive deploy() {
+        JavaArchive testJar = ShrinkWrap
+            .create(JavaArchive.class, "interceptorChangeFtAnnotation.jar")
+            .addClasses(InterceptorComponent.class, EarlyFtInterceptor.class, LateFtInterceptor.class, OrderQueueProducer.class)
+            .addAsManifestResource(new StringAsset("mp.fault.tolerance.interceptor.priority=3700"), "microprofile-config.properties")
+            .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+            .as(JavaArchive.class);
+
+        return ShrinkWrap.create(WebArchive.class, "interceptorChangeFtAnnotation.war")
+            .addAsLibrary(testJar);
+    }
+
+    /**
+     * This test validates the interceptors execution order after call a method
+     * annotated with Asynchronous FT annotation, using a queue type FIFO (first-in-first-out).
+     * The head of the queue is that element that has been on the queue the longest time.
+     * In this case is validating that the early interceptor is executed at first.
+     * @throws InterruptedException if the current thread was interrupted while waiting
+     * @throws ExecutionException if the computation threw an exception
+     */
+    @Test
+    public void testAsync() throws InterruptedException, ExecutionException {
+        Future<String> result = testInterceptor.asyncGetString();
+        assertEquals(result.get(), "OK");
+        String [] expectedOrder = {"asyncGetString","EarlyOrderFtInterceptor","LateOrderFtInterceptor"};
+        /*After changing the FT interceptor priority the first interceptor to be called must be FT interceptor*/
+        assertEquals(orderFactory.getOrderQueue().toArray(), expectedOrder);
+    }
+
+    @Test
+    public void testRetryInterceptors() {
+        try {
+            testInterceptor.serviceRetryA();
+            fail("Exception not thrown");
+        }
+        catch (Exception e) {
+            assertEquals(e.getMessage().trim(), "retryGetString failed");
+        }
+
+        String [] expectedOrder = {"serviceRetryA","EarlyOrderFtInterceptor","LateOrderFtInterceptor",
+            "serviceRetryA","EarlyOrderFtInterceptor","LateOrderFtInterceptor"};
```

Comment:
Unless I missed something, I was expecting this to be `{"EarlyOrderFtInterceptor","LateOrderFtInterceptor", "serviceRetryA", "EarlyOrderFtInterceptor","LateOrderFtInterceptor", "serviceRetryA"};

Reply:
Hello @Azquelt,

This is not the case because the idea of this the is to validate that if the FT priority is changed to be called at first must work.

Reply:
Right, but even if the FT interceptor is called first, the method body is still called at the end.

The difference I expect to with the FT interceptor being first is that the interceptors are called on each retry, rather than only once at the start.

Reply:
Ok so for no make sense this test because as a developer if i change the FT priority i expect some change in the order, i think is better wait after the implementation to add this test. What do you think?

Reply:
It looks like we have the following:
```
EarlyOrderFtInterceptor - priority 3800
LateOrderFtInterceptor - priority 3900
FT interceptor base - default priority 4010, but configured here to be 3700
```

If the FT interceptor is left at the **default priority**, I would expect the order to be "Early, Late, FT".

Therefore, in the order queue, I would expect to see `{"EarlyOrderFtInterceptor","LateOrderFtInterceptor", "asyncGetString", "asyncGetString"}`

If the FT interceptor is configured to **priority 3700**, I would then expect the order to be "FT, Early, Late".

Therefore, in the order queue, I would expect to see `{"EarlyOrderFtInterceptor","LateOrderFtInterceptor", "asyncGetString", "EarlyOrderFtInterceptor","LateOrderFtInterceptor", "asyncGetString"}`

The change is expected here because I now expect the interceptors to be called on each retry attempt.

Reply:
Ok @Azquelt Thanks for your clarification